src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -188,7 +188,74 @@ public function orWhereDoesntHave($relation, Closure $callback = null)
      * @param  mixed  $relations
      * @return $this
      */
-    public function withCount($relations)
+    public function withCount(...$relations)
+    {
+        return $this->withAggregate('count', ...$relations);
+    }
+
+    /**
+     * Add subselect queries to retrieve the minimum value of a column on the relations.
+     *
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withMin(...$relations)
+    {
+        return $this->withAggregate('min', ...$relations);
+    }
+
+    /**
+     * Add subselect queries to retrieve the maximum value of a column on the relations.
+     *
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withMax(...$relations)
+    {
+        return $this->withAggregate('max', ...$relations);
+    }
+
+    /**
+     * Add subselect queries to retrieve the sum the value of a column on the relations.
+     *
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withSum(...$relations)
+    {
+        return $this->withAggregate('sum', ...$relations);
+    }
+
+    /**
+     * Add subselect queries to retrieve the average the value of a column on the relations.
+     *
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withAvg(...$relations)
+    {
+        return $this->withAggregate('avg', ...$relations);
+    }
+
+    /**
+     * Alias for the "withAvg" method.
+     *
+     * @param  mixed  $relations
+     * @return mixed
+     */
+    public function withAverage(...$relations)
+    {
+        return $this->withAvg(...$relations);
+    }
+
+    /**
+     * Add subselect queries to retrieve the aggregate function of a column on the relations.
+     *
+     * @param  string $function
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withAggregate($function, $relations)
```

Comment:
What do you think about either typing this as `array $relations`, or using `...$relations`. Using `...$relations` would mean you can just do something like
```
if (is_array($relations[0])) {
    $relations = $relations[0];
}
```

I think that'd be a bit more clear than the `array_slice(func_get_args(), 1)` below.

Reply:
As I said in a previous comment, `$relations` is not always an array.

`array_slice(func_get_args(), 1)` gets all arguments except the first one, because I added that first parameter to know the aggregate function to apply